docs/design-ux/search.md

```diff
@@ -0,0 +1,35 @@
+# Search
+
++ Filename search -- `fzf` search via path (ctrl + l)
++ Full lockbook content search (ctrl + shift + f)
++ Text editor search (ctrl + f)
+
+# Filename search
+
+You hand core a string, core will hand you back a list of paths that this string matches. Can be integrated into
+quick-navigation of applications like it's done in the Linux application.
+
+# Lockbook Content Search
+
+Be able to quickly find files that match a certain criteria inside your lockbook. Not all the files in your Lockbook are
+guaranteed to be available on disk. And not every file type lends itself well to search-time scans.
+
+I propose we add a field to `FileMetadata` that contains `tags` or some other search term. Upon document writes these
+tags are updated. [Stop words](https://en.wikipedia.org/wiki/Stop_word) can be filtered out, in the future we can
+perform OCR on images and drawings. This field of `FileMetadata` gets encrypted and synced. When a user goes to perform
+a content search, all of these fields are scanned.
+
+# Text editor search
+
+Implementation will be upto the text editors in question, will perform similar to "Search this page" in your we browser.
+
+# Performance considerations
+
+My thoughts on how we'd go about improving the performance of these searches should that become an issue:
+
++ implement these with progress indication in mind. A communicative UI that's slow is better than a confusing UI that'
```

Comment:
I agree -- I think a lot of text editors I've used have sort of a "streaming search" that add new updates to the search results as it finds them.